### ISSUE TITLE
Extend the H-axis reduce split on TILE inputs to ttnn.max / ttnn.min

### ISSUE THREAD

**Issue by ign-msati** [tenstorrent/tt-metal#56103]
### Is your feature request related to a problem? Please describe

An H-axis reduce parallelizes only over output tile columns — `num_cols = NC * Wt` (`reduce_op_multi_core_h_program_factory.cpp`) — so a narrow-`W` tensor starves the grid no matter how tall it is. `(1, 1, 51264, 32)` gives `NC * Wt == 1`: one work unit, one core, 51264 rows serialized, 63 cores idle. That column count is computed the same way for every `ReduceOpMath`, so `ttnn.max` and `ttnn.min` sit on one core exactly like `ttnn.mean` did.

**Two H-axis splits now exist, and neither can reach `min`/`max`:**

- **#52922 / #54349** added the split for **TILE** inputs, scoped to `sum`/`mean`. The same `AVG`/`SUM` math gate excludes the `max`  and `min`

This issue tracks that follow-up. `min`/`max` on a tall, narrow-`W` tensor stay single-core on every path today, and **the tiled split is the only place this optimization can ever apply to them.**

### Describe the solution you'd like

Extend #54349's TILE H-axis split to **`ttnn.max` and `ttnn.min`**, reusing the two-stage machinery it already landed: stage 1 reduces `S` H-slices independently into partials, stage 2 collapses the slice axis.

**Already in place from #54349** — none of this needs to be rebuilt:

- `compute_h_slices` with per-path `min_ht_for_split` / `min_ht_per_slice` thresholds.
- Reader `num_h_slices` / `slice_Ht` compile args and the `(nc, slice, wt)` work-unit indexing; `{1, Ht}` keeps the un-split walk.
- `tile_h_split` in the multi-core-H factory: RM writer for the partials, `Ht = slice_Ht` on compute, global work-unit id on the reader.
- The device op admitting a TILE H-reduce that emits ROW_MAJOR partials, and Welford's opt-out.

### Additional context

  ```
import pytest
  import torch
  import ttnn
  from tests.ttnn.utils_for_testing import assert_equal
  
  pytestmark = pytest.mark.use_module_device
  
  
  @pytest.mark.parametrize("op", ["max", "min"])
  @pytest.mark.parametrize("dtype", [ttnn.bfloat16, ttnn.float32])
  # (N, 1, H, W) — reduced over the H axis; NC * Wt is 1..4, so the un-split path starves the grid.
  @pytest.mark.parametrize(
      "shape",
      [
          (1, 1, 51264, 32),
          (1, 1, 12544, 32),
          (1, 1, 3216, 128),
      ],
  )
  @pytest.mark.parametrize("spike", ["first", "middle", "last"])
  def test_select_h_axis(device, op, dtype, shape, spike):
      _, _, h, _ = shape
      row = {"first": 0, "middle": h // 2, "last": h - 1}[spike]
      torch_dtype = torch.float32 if dtype == ttnn.float32 else torch.bfloat16
      torch.manual_seed(0)
      torch_input = torch.zeros(shape, dtype=torch_dtype)
      torch_input[..., row, :] = 100.0 if op == "max" else -100.0
      torch_ref = (torch.amax if op == "max" else torch.amin)(torch_input, dim=-2, keepdim=True)
      tt_input = ttnn.from_torch(torch_input, dtype=dtype, layout=ttnn.TILE_LAYOUT, device=device)
      tt_fn = ttnn.max if op == "max" else ttnn.min
      output = ttnn.to_torch(tt_fn(tt_input, dim=-2, keepdim=True))
      # Selection is exact across a split — the partials pick a value rather than accumulate one.
      assert_equal(torch_ref, output)

```
